src/constants/fixed_bases.rs

```diff
@@ -278,78 +158,4 @@ impl FixedPoint<pallas::Affine> for ValueCommitV {
-fn test_lagrange_coeffs<C: CurveAffine>(base: C, num_windows: usize) {
-    let lagrange_coeffs = compute_lagrange_coeffs(base, num_windows);
-
-    // Check first 84 windows, i.e. `k_0, k_1, ..., k_83`
-    for (idx, coeffs) in lagrange_coeffs[0..(num_windows - 1)].iter().enumerate() {
-        // Test each three-bit chunk in this window.
-        for bits in 0..(1 << FIXED_BASE_WINDOW_SIZE) {
-            {
-                // Interpolate the x-coordinate using this window's coefficients
-                let interpolated_x = super::evaluate::<C>(bits, coeffs);
-
-                // Compute the actual x-coordinate of the multiple [(k+2)*(8^w)]B.
-                let point = base
-                    * C::Scalar::from_u64(bits as u64 + 2)
-                    * C::Scalar::from_u64(H as u64).pow(&[idx as u64, 0, 0, 0]);
-                let x = *point.to_affine().coordinates().unwrap().x();
-
-                // Check that the interpolated x-coordinate matches the actual one.
-                assert_eq!(x, interpolated_x);
-            }
-        }
-    }
-
-    // Check last window.
-    for bits in 0..(1 << FIXED_BASE_WINDOW_SIZE) {
-        // Interpolate the x-coordinate using the last window's coefficients
-        let interpolated_x = super::evaluate::<C>(bits, &lagrange_coeffs[num_windows - 1]);
-
-        // Compute the actual x-coordinate of the multiple [k * (8^84) - offset]B,
-        // where offset = \sum_{j = 0}^{83} 2^{3j+1}
-        let offset = (0..(num_windows - 1)).fold(C::Scalar::zero(), |acc, w| {
-            acc + C::Scalar::from_u64(2).pow(&[
-                FIXED_BASE_WINDOW_SIZE as u64 * w as u64 + 1,
-                0,
-                0,
-                0,
-            ])
-        });
-        let scalar = C::Scalar::from_u64(bits as u64)
-            * C::Scalar::from_u64(H as u64).pow(&[(num_windows - 1) as u64, 0, 0, 0])
-            - offset;
-        let point = base * scalar;
-        let x = *point.to_affine().coordinates().unwrap().x();
-
-        // Check that the interpolated x-coordinate matches the actual one.
-        assert_eq!(x, interpolated_x);
-    }
-}
-
-#[cfg(test)]
-// Test that the z-values and u-values satisfy the conditions:
-//      1. z + y = u^2,
-//      2. z - y is not a square
-// for the y-coordinate of each fixed-base multiple in each window.
-fn test_zs_and_us<C: CurveAffine>(base: C, z: &[u64], u: &[[[u8; 32]; H]], num_windows: usize) {
-    let window_table = compute_window_table(base, num_windows);
-
-    for ((u, z), window_points) in u.iter().zip(z.iter()).zip(window_table) {
-        for (u, point) in u.iter().zip(window_points.iter()) {
-            let y = *point.coordinates().unwrap().y();
-            let u = C::Base::from_bytes(u).unwrap();
-            assert_eq!(C::Base::from_u64(*z) + y, u * u); // allow either square root
-            assert!(bool::from((C::Base::from_u64(*z) - y).sqrt().is_none()));
-        }
-    }
 }
```

Comment:
Hmm, I'm -0 on removing this helper (which is ensuring that we have the correct hard-coded `Z` and `U` values). Instead we should expose `compute_window_tables` from `gadget::ecc::chip` and use it here. Once we make the `halo2_gadgets` split, it can be behind a `test-dependencies` feature flag.

Reply:
I agree with @str4d, except that I'm -100 (i.e. absolute veto) on removing these tests.

The problem is that without them, it's possible to backdoor the constants, as I pointed out in https://github.com/zcash/orchard/pull/43#discussion_r602737833 when they were originally added. (@ebfull convinced me that tests are sufficient rather than checking them on startup.)

Reply:
I've moved this to `ecc::chip::constants`, similarly to how `test_lagrange_coeffs`: https://github.com/zcash/orchard/pull/186/files#diff-08c8c700930e08d8a3478c6e1dbd727b5ab751a3514dbb8bf818b052ace8f8dbR180

Reply:
These tests have now been reintroduced. The `test_lagrange_coeffs` and `test_zs_and_us` functionality are exported from `ecc::chip::constants`. @daira said that it was unnecessary to add a `test-dependencies` feature flag since these are effectively part of the public API (we can't test Orchard without it).

Reply:
I agree we don't need that flag in `orchard`. What I meant is that, once `halo2_gadgets` exists, we have that flag _there_, and then in `orchard` we enable the flag in `dev-dependencies`, since we _only_ need these to test Orchard, and not to use it.